scenarios/aws/microVMs/microvms/libvirt.go

```diff
@@ -142,12 +142,13 @@ func (vm *VMCollection) SetupCollectionFilesystems(depends []pulumi.Resource) ([
 func (vm *VMCollection) SetupCollectionDomainConfigurations(depends []pulumi.Resource) ([]pulumi.Resource, error) {
 	var waitFor []pulumi.Resource
 
+	cpusAssigned := 0
 	for _, set := range vm.vmsets {
 		fs, ok := vm.fs[set.ID]
 		if !ok {
 			return nil, fmt.Errorf("failed to find filesystem for vmset %s", set.ID)
 		}
-		domains, err := GenerateDomainConfigurationsForVMSet(vm.instance.e.CommonEnvironment, vm.libvirtProviderFn, depends, &set, fs)
+		domains, err := GenerateDomainConfigurationsForVMSet(vm.instance.e.CommonEnvironment, vm.libvirtProviderFn, depends, &set, fs, &cpusAssigned)
```

Comment:
Why not multiple return value for `cpusAssigned`? Using `*int` like this is not idiomatic Go.

Reply:
I added some suggestions to show you what I mean. There are still more changes needed, but that is the idea.

Reply:
I was doing that [here](https://github.com/DataDog/test-infra-definitions/pull/526/commits/2da089786a5ca85b6ffd951068d79d56178e5d0d) but it resulted in a compiler error.

I guess I need to declare all the return values separately, otherwise the compiler creates a new local variable for `cpuAssigned`

Reply:
or don't use `:=` on that line and declare the other variables above

Reply:
Yeah thats what what I meant